src/hoi4_world/military/convoy_distributor.cpp

```diff
@@ -0,0 +1,54 @@
+#include "src/hoi4_world/military/convoy_distributor.h"
+
+namespace hoi4
+{
+
+void ConvoyDistributor::CalculateStateWeights(const vic3::World& source_world)
+{
+   const auto& states = source_world.GetStates();
+   const auto& buildings = source_world.GetBuildings();
+
+   vic3_total_ = 0;
+   vic3_map_ = {};
+   for (const auto [id, state]: states)
+   {
+      const auto building = buildings.GetBuildingInState(id, vic3::BuildingType::Port);
+      if (!building)
+      {
+         continue;
+      }
+      const auto& methods = building->GetProductionMethods();
+      float pmWeight = 0;
+      for (const auto pm: methods)
+      {
+         if (!pm_weights_.contains(pm))
+         {
+            continue;
```

Comment:
You should log when the port doesn't have a handled production method. And have a test case for it too.

Reply:
Added the logging. There's a `dummy` entry in some of the test cases that exercises this code.